src/WarcraftLegacies.Source/Quests/Goblin/QuestRatchet.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System.Collections.Generic;
+using MacroTools.ControlPointSystem;
+using MacroTools.Extensions;
+using MacroTools.FactionSystem;
+using MacroTools.LegendSystem;
+using MacroTools.ObjectiveSystem.Objectives.ControlPointBased;
+using MacroTools.ObjectiveSystem.Objectives.FactionBased;
+using MacroTools.ObjectiveSystem.Objectives.LegendBased;
+using MacroTools.ObjectiveSystem.Objectives.UnitBased;
+using MacroTools.QuestSystem;
+using static War3Api.Common;
+
+
+namespace WarcraftLegacies.Source.Quests.Goblin
+{
+  /// <summary>
+  /// The Goblins can acquire Ratchet.
+  /// </summary>
+  public sealed class QuestRatchet : QuestData
+  {
+    private readonly List<unit> _rescueUnits;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="QuestRatchet"/>.
+    /// </summary>
+    /// <param name="noggenfogger">Must be brought somewhere to complete the quest.</param>
+    public QuestRatchet(LegendaryHero noggenfogger) : base("Ratchet Port",
+      "The port of Ratchet would be a great expansion of our trade empire.",
+      @"ReplaceableTextures\CommandButtons\BTNGoblinShop2.blp")
+    {
+      AddObjective(new ObjectiveControlPoint(ControlPointManager.Instance.GetFromUnitType(Constants.UNIT_N0A6_RATCHET)));
+      AddObjective(new ObjectiveLegendInRect(noggenfogger, Regions.Ratchet_Unlock_1, "Ratchet"));
```

Comment:
Gazlowe founded Ratchet in canon, not Noggenfogger, so I think it's odd to see Noggenfogger as the forced founder here. Could we change this to the `ObjectiveAnyUnitInRect` objective with the `heroOnly` parameter set to true? That way Gazlowe could _technically_ found it, even if Noggenfogger probably always will.